docs/source/usecases/VirtualFunctions.md

```diff
@@ -111,6 +111,60 @@ deviceInstance->setAField(someHostValue); // set some field on the host
 
 This is the solution that the code teams we have talked to have said is the most productive way to solve the problem.
 
+## But what if I do not really need the V-Tables on the device side?
+Consider the following example which calls the `virtual operator()` on the device from a pointer of derived class type.
+One might think this should work because no V-Table lookup on the device is neccessary.
+```c++
+struct Interface
+{
+    KOKKOS_DEFAULTED_FUNCTION
+    virtual ~Interface() = default;
+
+    KOKKOS_FUNCTION
+    virtual void operator()( const size_t) const = 0;
+};
+
+struct Implementation : public Interface
+{
+    KOKKOS_FUNCTION
+    void operator()(const size_t i) const override
+    { ... }
+
+    void apply(){
+        Kokkos::parallel_for("myLoop",10,
+            KOKKOS_CLASS_LAMBDA (const size_t i) { this->operator()(i); }
+        );
+    }
+};
+
+int main ()
+{
+    ... 
+    auto implementationPtr = std::make_shared<Implementation>();
+    implementationPtr->apply();
+}
```

Comment:
I would prefer the example to actually compile.

Reply:
done in [6ba9f8](https://github.com/kokkos/kokkos-core-wiki/pull/162/commits/6ba9f8ee1ce63d6c7ec42203bbf02f3cd1d8acab)